Src/Support/Google.Apis.Auth/OAuth2/FileSourcedExternalAccountCredential.cs

```diff
@@ -0,0 +1,139 @@
+    /// https://google.aip.dev/auth/4117.
+    /// </summary>
+    public sealed class FileSourcedExternalAccountCredential : ExternalAccountCredential, IGoogleCredential
+    {
+        new internal class Initializer : ExternalAccountCredential.Initializer
+        {
+            /// <summary>
+            /// The file from which to obtain the subject token.
+            /// </summary>
+            internal string SubjectTokenFilePath { get; }
+
+            /// <summary>
+            /// If set, the subject token file content will be parsed as JSON and the
+            /// value in the field with name <see cref="SubjectTokenJsonFieldName"/>
+            /// will be returned as the subject token.
+            /// </summary>
+            internal string SubjectTokenJsonFieldName { get; set; }
+
+            internal Initializer(string tokenUrl, string audience, string subjectTokenType, string subjectTokenFilePath)
+                : base(tokenUrl, audience, subjectTokenType) => SubjectTokenFilePath = subjectTokenFilePath;
+
+            internal Initializer(Initializer other) : base(other)
+            {
+                SubjectTokenFilePath = other.SubjectTokenFilePath;
+                SubjectTokenJsonFieldName = other.SubjectTokenJsonFieldName;
+            }
+
+            internal Initializer(FileSourcedExternalAccountCredential other) : base(other)
+            {
+                SubjectTokenFilePath = other.SubjectTokenFilePath;
+                SubjectTokenJsonFieldName = other.SubjectTokenJsonFieldName;
+            }
+        }
+
+        /// <summary>
+        /// The file path from which to obtain the subject token.
+        /// </summary>
+        public string SubjectTokenFilePath { get; }
+
+        /// <summary>
+        /// If set, the subject token file content will be parsed as JSON and the
+        /// value in the field with name <see cref="SubjectTokenJsonFieldName"/>
+        /// will be returned as the subject token.
+        /// </summary>
+        public string SubjectTokenJsonFieldName { get; }
+
+        internal FileSourcedExternalAccountCredential(Initializer initializer) : base(initializer)
+        {
+            SubjectTokenFilePath = initializer.SubjectTokenFilePath;
+            SubjectTokenJsonFieldName = initializer.SubjectTokenJsonFieldName;
+        }
+
+        /// <inheritdoc/>
+        private protected override GoogleCredential WithoutImpersonationConfigurationImpl() =>
+            ServiceAccountImpersonationUrl is null
+            ? new GoogleCredential(this)
+            : new GoogleCredential(new FileSourcedExternalAccountCredential(new Initializer(this)
+            {
+                ServiceAccountImpersonationUrl = null
+            }));
+
+        /// <inheritdoc/>
+        protected override Task<string> GetSubjectTokenAsync(CancellationToken taskCancellationToken)
+        {
+            var fileContent = File.ReadAllText(SubjectTokenFilePath);
```

Comment:
Do we definitely want to read the file here, rather than in the constructor? What's the expected behavior if this file changes over time?

Reply:
The file content can change at any time, so I think it's correct to read the file here.